src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -62,6 +63,16 @@ public function getPath()
         return $this->path;
     }
 
+    public function setLocales(array $locales)
```

Comment:
Is there a reason to have this setter?

Reply:
This is how annotations hydrate itself.

Reply:
Small nitpicking, it's how this particular annotation behaves. Sadly there's no real standard for it. Doctrine for example, uses public properties from what I can tell. Other annotations might use a k=>v storage. 

For this particular annotation it's required to have a setter as @frankdejonge mentioned

Reply:
in theory we could reuse `set/getPath()` supporting both string and array API. Not sure if that simplifies the code in any way..

Reply:
@ro0NL could be, but we've handled it in the constructor now, which enables us to add type check for returns and make the "special" case more explicit. I think especially making the case explicit adds a lot of value because it's more clear.